rest/route/repo.go

```diff
@@ -57,9 +66,320 @@ func (h *repoIDGetHandler) Run(ctx context.Context) gimlet.Responder {
 	}
 	repoModel.Variables = *repoVars
 
-	if repoModel.Aliases, err = h.sc.FindProjectAliases("", repo.Id); err != nil {
+	if repoModel.Aliases, err = h.sc.FindProjectAliases("", repo.Id, nil); err != nil {
 		return gimlet.MakeJSONErrorResponder(err)
 	}
 
 	return gimlet.NewJSONResponse(repoModel)
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+// PATCH /rest/v2/repos/{repo_id}
+
+type repoIDPatchHandler struct {
+	repoName       string
+	originalRepo   *dbModel.RepoRef
+	newRepoRef     *dbModel.RepoRef
+	requestRepoRef *model.APIProjectRef
+	user           *user.DBUser
+
+	sc       data.Connector
+	settings *evergreen.Settings
+}
+
+func makePatchRepoByID(sc data.Connector, settings *evergreen.Settings) gimlet.RouteHandler {
+	return &repoIDPatchHandler{
+		sc:       sc,
+		settings: settings,
+	}
+}
+
+func (h *repoIDPatchHandler) Factory() gimlet.RouteHandler {
+	return &repoIDPatchHandler{
+		sc:       h.sc,
+		settings: h.settings,
+	}
+}
+
+func (h *repoIDPatchHandler) Parse(ctx context.Context, r *http.Request) error {
+	h.repoName = gimlet.GetVars(r)["repo_id"]
+	h.user = MustHaveUser(ctx)
+	body := util.NewRequestReader(r)
+	defer body.Close()
+	b, err := ioutil.ReadAll(body)
+	if err != nil {
+		return errors.Wrap(err, "Argument read error")
+	}
+
+	// get the old repo
+	h.originalRepo, err = dbModel.FindOneRepoRef(h.repoName)
+	if err != nil {
+		return err
+	}
+	if h.originalRepo == nil {
+		return errors.Errorf("repo '%s' doesn't exist", h.repoName)
+	}
+
+	h.requestRepoRef = &model.APIProjectRef{}
+	if err = h.requestRepoRef.BuildFromService(h.originalRepo.ProjectRef); err != nil {
+		return errors.Wrap(err, "API error converting from model.ProjectRef to model.APIProjectRef")
+	}
+
+	// erase contents so requestRepoRef will only be populated with new elements for these fields
+	h.requestRepoRef.Admins = nil
+	h.requestRepoRef.GitTagAuthorizedUsers = nil
+	h.requestRepoRef.GitTagAuthorizedTeams = nil
+	if err = json.Unmarshal(b, h.requestRepoRef); err != nil {
+		return errors.Wrap(err, "API error while unmarshalling JSON")
+	}
+
+	// read the new changes onto it
+	i, err := h.requestRepoRef.ToService()
+	if err != nil {
+		return errors.Wrap(err, "API error converting from model.APIProjectRef to model.ProjectRef")
+	}
+	pRef, ok := i.(*dbModel.ProjectRef)
+	if !ok {
+		return errors.Errorf("unexpected type %T for model.ProjectRef", i)
+	}
+	h.newRepoRef = &dbModel.RepoRef{ProjectRef: *pRef}
+	return nil
+}
+
+func (h *repoIDPatchHandler) Run(ctx context.Context) gimlet.Responder {
+
+	before, err := h.sc.GetProjectSettingsEvent(&h.newRepoRef.ProjectRef)
+	if err != nil {
+		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "Error getting ProjectSettingsEvent before update for repo '%s'", h.repoName))
+	}
+
+	catcher := grip.NewSimpleCatcher()
+	catcher.Add(h.newRepoRef.ValidateOwnerAndRepo(h.settings.GithubOrgs))
```

Comment:
this should maybe go in the parse function

Reply:
Most of what's in Run is validating the input we've been given, so I think it makes sense as part of that logic